server/src/main/java/org/opensearch/tasks/TaskResourceStats.java

```diff
@@ -32,27 +35,41 @@
  */
 public class TaskResourceStats implements Writeable, ToXContentFragment {
     private final Map<String, TaskResourceUsage> resourceUsage;
+    private final TaskThreadUsage threadUsage;
 
-    public TaskResourceStats(Map<String, TaskResourceUsage> resourceUsage) {
+    public TaskResourceStats(Map<String, TaskResourceUsage> resourceUsage, TaskThreadUsage threadUsage) {
         this.resourceUsage = Objects.requireNonNull(resourceUsage, "resource usage is required");
+        this.threadUsage = Objects.requireNonNull(threadUsage, "thread usage is required");
     }
 
     /**
      * Read from a stream.
      */
     public TaskResourceStats(StreamInput in) throws IOException {
         resourceUsage = in.readMap(StreamInput::readString, TaskResourceUsage::readFromStream);
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            threadUsage = TaskThreadUsage.readFromStream(in);
+        } else {
+            threadUsage = null;
```

Comment:
```suggestion
            // Initialize TaskThreadUsage in case it is not found in mixed cluster case
            threadUsage = new TaskThreadUsage(0, 0);
```